src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3655,6 +3658,13 @@ def _fill_defaults_for_pod_identity_exceptions(pod_identity_exceptions):
             exc.pod_labels = dict()
 
 
+def _fill_defaults_for_pod_identity_profile(pod_identity_profile):
+    if not pod_identity_profile:
+        return
+
+    _fill_defaults_for_pod_identity_exceptions(pod_identity_profile.user_assigned_identity_exceptions)
```

Comment:
The PR looks good to me. Could you please add unit test cases for these internal functions?

- _fill_defaults_for_pod_identity_profile
- _fill_defaults_for_pod_identity_exceptions
- _update_addon_pod_identity

Reply:
`_update_addon_pod_identity` is not part of current pr change, I will add unit tests in follow up pr. Added unit tests for the `_fill_defaults` helper methods.